include/deal.II/distributed/tria.h

```diff
@@ -815,6 +815,15 @@ namespace parallel
       const std::vector<types::global_dof_index> &
       get_coarse_cell_to_p4est_tree_permutation() const;
 
+      /**
+       * This returns a pointer to the internally stored p4est object (of type p4est_t or p8est_t depending on @p dim).
+       *
+       * Warning: if you modify the p4est, internal data structures can become
+       * inconsistent.
```

Comment:
Which is, I think, exactly what you should return a `const` object :-)

Reply:
The second I need to use anything in the p4est API, I would need to cast const away though. The APIs are not const-correct . :-(

Reply:
I guess that is still cleaner, so I will do the change...